### PR TITLE
feat: Add custom user agent enhancements for Predictions

### PR DESCRIPTION
**Pull request by stocaaro**
#### Description of changes
Add Predictions category with specific actions for the custom user agent in all predictions call *except "convert" for voice to text (transcription), which uses websocket streams and doesn't currently support custom user agent.*

#### Description of how you validated changes

Unit tests added to cover user agent detection in sdk client. I'll be going through manual testing for predictions before this is merged to main.

#### Checklist
- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/predictions/src/Providers/AmazonAIInterpretPredictionsProvider.ts [hunk 1]
```diff
@@ -48,10 +55,13 @@ export class AmazonAIInterpretPredictionsProvider extends AbstractInterpretPredi
 			const { text: { source: { language = undefined } = {} } = {} } = ({} =
 				input as any); // language is only required for specific interpret types
 
-			const comprehendClient = new ComprehendClient({
+			this.comprehendClient = new ComprehendClient({
```
Comment (AllanZhengYP) [inline packages/predictions/src/Providers/AmazonAIInterpretPredictionsProvider.ts:58]: Nice to see this improvement!

File: packages/predictions/src/Providers/AmazonAIInterpretPredictionsProvider.ts [hunk 2]
```diff
@@ -260,13 +252,16 @@ export class AmazonAIInterpretPredictionsProvider extends AbstractInterpretPredi
...
 		try {
 			const detectDominantLanguageCommand = new DetectDominantLanguageCommand(
 				params
 			);
-			const data = await comprehend.send(detectDominantLanguageCommand);
-			const { Languages: [{ LanguageCode }] = [''] } = ({} = data || {});
+			const data = await this.comprehendClient.send(
+				detectDominantLanguageCommand
+			);
+			const x = data?.Languages?.[0];
```
Comment (AllanZhengYP) [inline packages/predictions/src/Providers/AmazonAIInterpretPredictionsProvider.ts:263]: Extra line?

Comment (stocaaro) [reply packages/predictions/src/Providers/AmazonAIInterpretPredictionsProvider.ts]: Yep. Was just debugging a ts issue that having the client typed introduced and forgot to remove my syntax check line. Thx for catching!

File: packages/predictions/src/Providers/AmazonAIInterpretPredictionsProvider.ts (hunk 1 above)
Comment (stocaaro) [reply packages/predictions/src/Providers/AmazonAIInterpretPredictionsProvider.ts:58]: Idk about improvement. Its consistent with the other providers in predictions now, and by saving the client to the instance it makes testing much more straightforward.

**Review by erinleigh90 — APPROVED** [review]
LGTM

**Pull request merged**